src/test/java/competition/subsystems/hanger/HangerSubsystemTest.java

```diff
@@ -0,0 +1,49 @@
+package competition.subsystems.hanger;
+
+import static org.junit.Assert.assertEquals;
+
+import org.junit.Test;
+
+import competition.BaseCompetitionTest;
+import competition.subsystems.hanger.HangerSubsystem;
+
+public class HangerSubsystemTest extends BaseCompetitionTest {
+    @Test
+    public void testHangerExtendHanger() {
+        HangerSubsystem hanger = this.injector.getInstance(HangerSubsystem.class);
+        hanger.extendHanger();
+        assertEquals(1, hanger.hangerMotor.getMotorOutputPercent(), 0.001);
+    }
+
+    @Test
+    public void testRetractHanger() {
+        HangerSubsystem hanger = this.injector.getInstance(HangerSubsystem.class);
+        hanger.retractHanger();
+        assertEquals(-1, hanger.hangerMotor.getMotorOutputPercent(), 0.001);
+    }
+
+    @Test
+    public void testHangerGrabClamp() {
+        HangerSubsystem hanger = this.injector.getInstance(HangerSubsystem.class);
+        hanger.grabClamp();
+        assertEquals(1, hanger.hangerMotor.getMotorOutputPercent(), 0.001);
+    }
+
+    @Test
+    public void testHangerReleaseClamp() {
+        HangerSubsystem hanger = this.injector.getInstance(HangerSubsystem.class);
+        hanger.releaseClamp();
+        assertEquals(-1, hanger.hangerMotor.getMotorOutputPercent(), 0.001);
+    }
+
+    @Test
+    public void testHangerStop() {
+        HangerSubsystem hanger = this.injector.getInstance(HangerSubsystem.class);
+        hanger.extendHanger();
```

Comment:
Basic pattern for this thing:
1) Make it extend
2) Verify (via assert) it is extending
3) Make it stop
4) Verify (via assert) it has stopped.